File: doc-examples/jdbc-example-records-java/src/main/java/example/CourseRating.java
package example;

import io.micronaut.core.annotation.Nullable;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Relation;

@MappedEntity
public record CourseRating(@Id @GeneratedValue @Nullable
                           Long id,
                           @Relation(Relation.Kind.MANY_TO_ONE)
                           Student student,
                           @Relation(Relation.Kind.MANY_TO_ONE)
                           Course course,
                           int rating) {

    public CourseRating(Student student, Course course, int rating) {
        this(null, student, course, rating);
    }

}

File: doc-examples/mongo-reactive-example-kotlin/src/main/kotlin/example/CourseRating.kt
package example

import io.micronaut.core.annotation.Nullable
import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Relation
import org.bson.types.ObjectId

@MappedEntity
data class CourseRating(
        @field:Id @GeneratedValue @Nullable
        val id: ObjectId?,
        @Relation(Relation.Kind.MANY_TO_ONE)
        val student: Student,
        @Relation(Relation.Kind.MANY_TO_ONE)
        val course: Course,
        val rating: Int
) {
    constructor(student: Student, course: Course, rating: Int) : this(null, student, course, rating)
}